### PR TITLE
feat: rename crp conditions

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes

Rename workCreated to workSynchronized

Fixes #

I have:

- [X] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: apis/placement/v1beta1/clusterresourceplacement_types.go [hunk 1]
```diff
@@ -758,24 +758,7 @@ const (
 	// Its condition status can be one of the following:
 	// - "True" means we have successfully scheduled the resources to satisfy the placement requirement.
 	// - "False" means we didn't fully satisfy the placement requirement. We will fill the Message field.
-	// TODO, use "Scheduled" instead.
-	ResourceScheduledConditionType ResourcePlacementConditionType = "ResourceScheduled"
```
Comment (ryanzhang-oss) [inline apis/placement/v1beta1/clusterresourceplacement_types.go:762]: how come no existing code use this?

File: apis/placement/v1beta1/clusterresourceplacement_types.go [hunk 2]
```diff
@@ -758,24 +758,7 @@ const (
...
-	// the latest.
-	// - "False" means we have not created the latest corresponding work(s) or updated the existing work(s) to the latest
-	// yet.
-	// There are few possibilities:
-	// - In the processing state
-	// - Rollout controller has decided not to create or update the resources in this cluster for now to honor the
-	// rollout strategy configurations specified in the placement.
-	// - Work is not created/updated because of the unknown reasons.
-	// To be deprecated, it will be replaced by RolloutStarted and WorkCreated conditions.
-	ResourceWorkSynchronizedConditionType ResourcePlacementConditionType = "WorkSynchronized"
```
Comment (ryanzhang-oss) [inline apis/placement/v1beta1/clusterresourceplacement_types.go:778]: same

Comment (zhiying-lin) [reply apis/placement/v1beta1/clusterresourceplacement_types.go:778]: duplicate definition of line 781

File: apis/placement/v1beta1/clusterresourceplacement_types.go (hunk 1 above)
Comment (zhiying-lin) [reply apis/placement/v1beta1/clusterresourceplacement_types.go:762]: These conditions are info only and validated by tests.

To fix the rp side e2e tests, we just need to upgrade the fleet go mod.
https://msazure.visualstudio.com/CloudNativeCompute/_search?action=contents&text=place_resources_with_pickfixed_crp&type=code&lp=code-Project&filters=ProjectFilters%7BCloudNativeCompute%7DRepositoryFilters%7Baks-rp%7D&pageSize=25&result=DefaultCollection/CloudNativeCompute/aks-rp/GBmaster//test/e2e/suites/pkg/handlerk8s/handler_fleet_workload.go

or we can leave there and fix the rp e2e tests when enabling the new conditions.

i'm fine with either way.

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**